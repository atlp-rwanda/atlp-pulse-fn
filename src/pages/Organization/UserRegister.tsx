/* eslint-disable */
import { useApolloClient, useLazyQuery, useMutation } from '@apollo/client';
import React, { useContext, useState, useEffect } from 'react';
import { useForm } from 'react-hook-form';

import { useTranslation } from 'react-i18next';
import { FaGoogle, FaRegEnvelope, FaRegEye } from 'react-icons/fa';
import { FiEyeOff } from 'react-icons/fi';
import { MdLockOutline } from 'react-icons/md';
import { Link, useLocation, useNavigate } from 'react-router-dom';
import { toast } from 'react-toastify';
import ButtonLoading from '../../components/ButtonLoading';
import Button from '../../components/Buttons';
import { UserContext } from '../../hook/useAuth';
import useDocumentTitle from '../../hook/useDocumentTitle';
import { SIGN_UP_MUTATION, GET_SIGNUP_ORGANIZATION } from './Mutations';
import ControlledSelect from '../../components/ControlledSelect';
import jwt_decode from 'jwt-decode';

const Signup = () => {
  const token: any = window.location.href.substring(
    window.location.href.lastIndexOf('/') + 1,
  );
  const originalToken: any = token.replaceAll('*', '.');
  useDocumentTitle('Login');

  const { t } = useTranslation();
  const [passwordShown, setPasswordShown] = useState(false);
  const [buttonLoading, setButtonLoading] = useState(false);
  const tooglePassword =
    /* istanbul ignore next */
    () => {
      /* istanbul ignore next */
      setPasswordShown(!passwordShown);
    };
  const {
    register,
    handleSubmit,
    formState: { errors },
    setError,
    control,
    controller,
    setValue,
  }: any = useForm();

  const { UserSignup } = useContext(UserContext);
  const navigate = useNavigate();
  const { state } = useLocation();
  const [Signup, { loading }] = useMutation(SIGN_UP_MUTATION);

  const [getOrganizationName] = useLazyQuery(GET_SIGNUP_ORGANIZATION, {
    variables: {
      orgToken: originalToken,
    },

    /* istanbul ignore next */
    onError: (err) => {
      /* istanbul ignore next */
      if (err.message === 'expired organization token') {
        toast.error('Your signup link has expired');
        navigate('/');
      } else {
        navigate('/pageNotFound');
      }
    },
  });
  const [SignupUser] = useMutation(SIGN_UP_MUTATION, {
    /* istanbul ignore next */
    /* istanbul ignore next */
    onCompleted: (data) => {
      /* istanbul ignore next */
      setTimeout(() => {
        setButtonLoading(false);
        toast.success('You have successfully created an account!');
        navigate('/register-successful');
      }, 2000);
    },
    /* istanbul ignore next */
    onError: (err) => {
      /* istanbul ignore next */
      setTimeout(() => {
        setButtonLoading(false);
        toast.error(err.message);
      }, 1000);
    },
  });

  /* istanbul ignore next */
  const isDateOfBirthValid = (dateOfBirth: string) => {
    const currentDate = new Date();
    const dob = new Date(dateOfBirth);

    // Calculate the age difference
    const ageDifference = currentDate.getTime() - dob.getTime();

    // Calculate age based on milliseconds
    const age = Math.floor(ageDifference / (1000 * 60 * 60 * 24 * 365));

    // Check if the user is at least 18 years old
    return age >= 18;
  };

  const commonPasswords = [
    123456,
    'password',
    12345678,
    'qwerty',
    123456789,
    12345,
    1234,
    111111,
    1234567,
    'dragon',
    123123,
    'baseball',
    'abc123',
    'football',
    'monkey',
    'letmein',
    696969,
    'shadow',
    'master',
    666666,
    'qwertyuiop',
    123321,
    'mustang',
    1234567890,
    'michael',
    654321,
    'pussy',
    'superman',
    '1qaz2wsx',
    7777777,
    'fuckyou',
    121212,
    '000000',
    'qazwsx',
    '123qwe',
    'killer',
    'trustno1',
    'jordan',
    'jennifer',
    'zxcvbnm',
    'asdfgh',
    'hunter',
    'buster',
    'soccer',
    'harley',
    'batman',
    'andrew',
    'tigger',
    'sunshine',
    'iloveyou',
    'fuckme',
    2000,
    'charlie',
    'robert',
    'thomas',
    'hockey',
    'ranger',
    'daniel',
    'starwars',
    'klaster',
    112233,
    'george',
    'asshole',
    'computer',
    'michelle',
    'jessica',
    'pepper',
    1111,
    'zxcvbn',
    555555,
    11111111,
    131313,
    'freedom',
    777777,
    'pass',
    'fuck',
    'maggie',
    159753,
    'aaaaaa',
    'ginger',
    'princess',
    'joshua',
    'cheese',
    'amanda',
    'summer',
    'love',
    'ashley',
    6969,
    'nicole',
    'chelsea',
    'biteme',
    'matthew',
    'access',
    'yankees',
    987654321,
    'dallas',
    'austin',
    'thunder',
    'taylor',
    'matrix',
  ];
  const isStrongPassword = (password: any) => {
    // Check if the password length is at least 8 characters
    if (password.length < 8) {
      return false;
    }
    if (commonPasswords.includes(password.toLowerCase())) {
      return false;
    }
    if (commonPasswords.includes(password.toUpperCase())) {
      return false;
    }
    if (!/\d/.test(password)) {
      return false;
    }

    // Check if the password contains at least one special character
    if (!/[!@#$%^&*(),.?":{}|<>]/.test(password)) {
      return false;
    }
    return true;
  };
  /* istanbul ignore next */
  const onSubmit = async (userInput: any) => {
    const isAdult = isDateOfBirthValid(userInput.dateOfBirth);
    if (!isAdult) {
      toast.error('You must be at least 18 years old to sign up.');
      return;
    }
    const isPasswordStrong = isStrongPassword(userInput.password);
    if (!isPasswordStrong) {
      toast.error(
        'Password is too weak or too common. Please choose a stronger password.',
      );
      return;
    }

    setButtonLoading(true);
    setTimeout(async () => {
      try {
        await SignupUser({
          variables: {
            email: userInput.email,
            password: userInput.password,
            firstName: userInput.firstName,
            lastName: userInput.lastName,
            dateOfBirth: userInput.dateOfBirth,
            gender: userInput.gender.value,
            orgToken: originalToken,
          },
        });

        return;
      } catch (error: any) {
        toast.error(error.message);
      }
    }, 2000);
  };

  /* istanbul ignore next */
  const options: any = [
    { value: 'male', label: 'male' },
    { value: 'female', label: 'female' },
  ];

  useEffect(() => {
    interface TokenPayload {
      email: string;
    }
    // Get the email from the token and set it as the initial value for the email input field
    const getEmailFromToken = async () => {
      let decodedToken: TokenPayload = await jwt_decode(originalToken);
      const emailFromToken = decodedToken.email;

      // Set the email value in the email input field
      setValue('email', emailFromToken);
    };
    getOrganizationName();
    getEmailFromToken();
  }, [setValue]);

  // checking current logged in user
  /* istanbul ignore next */
  useEffect(() => {
    if (localStorage.getItem('auth_token')) navigate('/dashboard');
  }, []);

  return (
    <div className="md:flex md:flex-col md:items-center md:justify-center w-full grow text-center py-2 dark:bg-dark-bg bg-gray-100  sm:flex sm:flex-row sm:items-center sm:justify-center">
      <div className="w-full md:w-3/5 md:p-5 sm:w-full sm:p-2 dark:bg-dark-bg dark:rounded-none ">
        <div className="py-10 sm:py-8 w-full m-auto  md:shadow-xl rounded-xl">
          <h2 className="text-2xl font-bold text-primary dark:text-dark-text-fill ">
            {t('Sign up using')}
          </h2>
          <div className="border-2 w-10 bg-primary border-primary inline-block mb-2" />
          {/* <div className="flex justify-center my-2">
            <a
              href="#link"
              className="border-2 border-gray-200 rounded-full p-3 mx-1"
            >
              <FaGoogle className="text-sm dark:text-white" />
            </a>
          </div> */}
          <p className="text-gray-400 my-3 dark:text-dark-text-fill ">
            {t('use your email account')}
          </p>
          <div className="flex flex-col items-center">
            <form
              action="#none"
              className="flex flex-col gap-2"
              onSubmit={handleSubmit(onSubmit)}
              data-testid="loginForm"
            >
              <div className="flex justify-between flex-col md:flex-row gap-2">
                <div className="bg-gray-100 border border-primary rounded p-2 flex justify-between  dark:bg-dark-bg w-full mr-2">
                  <input
                    data-testid="fName"
                    type="name"
                    placeholder={t('First Name')}
                    {...register('firstName', {
                      required: 'First Name is required',
                    })}
                    className="bg-gray-100 outline-none text-sm flex-1 text-gray-400 dark:border-white dark:bg-dark-bg dark:text-white "
                  />
                </div>
                <div className="bg-gray-100 border border-primary rounded p-2 flex justify-between dark:bg-dark-bg w-full">
                  <input
                    data-testid="lName"
                    type="name"
                    placeholder={t('Last Name')}
                    {...register('lastName', {
                      required: 'Last Name is required',
                    })}
                    className="bg-gray-100 outline-none text-sm flex-1 text-gray-400 dark:border-white dark:bg-dark-bg dark:text-white "
                  />
                </div>
              </div>
              <div className="flex justify-between flex-col md:flex-row gap-2">
                <div className="grouped-input flex items-center  w-full  rounded-md mr-2">
                  <input
                    type="date"
                    className="text-gray-400 border border-primary py-2 dark:bg-dark-bg rounded outline-none px-5 font-sans text-xs w-full h-[38px] "
                    id="date-placeholder"
                    max={new Date().toISOString().split('T')[0]}
                    data-placeholder="Date of birth"
                    required
                    {...register('dateOfBirth', {
                      valueAsDate: true,
                      required: `${t('The BirthDate is required')}`,
                    })}
                  />
                </div>

                <ControlledSelect
                  placeholder={t('Select gender')}
                  register={{
                    control,
                    name: 'gender',
                    rules: {
                      required: `${t('Gender is required')}`,
                    },
                  }}
                  options={options}
                />
              </div>
              <div className="bg-gray-100 w-70 p-2 flex justify-between  dark:bg-dark-bg border border-primary rounded xs:w-full gap-2">
                <FaRegEnvelope className="text-gray-400 " />
                <input
                  data-testid="email"
                  type="email"
                  {...register('email', { required: 'Email is required' })}
                  placeholder={t('Email')}
                  className="bg-gray-100 outline-none text-sm flex-1 text-gray-400 dark:border-white dark:bg-dark-bg dark:text-white  "
                  readOnly
                />
              </div>
              {errors.email && (
                <div className="text-left  pl-4">
                  <small className="text-red-600">{errors.email.message}</small>
                </div>
              )}

              <div className="bg-gray-100  p-2 flex items-center border border-primary rounded   dark:bg-dark-bg xs:w-full md:w-70 gap-2">
                <MdLockOutline className="text-gray-400 " />
                <input
                  data-testid="password"
                  type={passwordShown ? 'text' : 'password'}
                  {...register('password', {
                    required: 'Password is required',
                  })}
                  placeholder={t('Password')}
                  className="bg-gray-100 outline-none text-sm flex-1 text-gray-400 dark:border-white dark:bg-dark-bg dark:text-white"
                />
                <div className="text-gray-400 cursor-pointer onClick= {()=> handleShowPassword}">
                  {passwordShown ? (
                    <FaRegEye onClick={tooglePassword} />
                  ) : (
                    <FiEyeOff onClick={tooglePassword} />
                  )}
                </div>
              </div>
              {errors.password && (
                <div className="text-left mb-1 pl-4">
                  <small className="text-red-600">
                    {errors.password.message}
                  </small>
                </div>
              )}
              <div className="flex w-64  rounded mb-5 mt-5 items-center ">
                <div className=" text-sm mr-1">
                  {t('Already have an account?')}
                </div>
                <Link to="/login/org" className="text-xs text-primary ">
                  {t('Log in')}
                </Link>
              </div>
              <div className="w-full justify-center">
                {loading ? (
                  <ButtonLoading style={'rounded-full inline-block'} />
                ) : (
                  <Button
                    type="submit"
                    variant="transparentbtn"
                    size="md"
                    style="border-2 hover:bg-primary inline-block rounded-full lg:px-12 lg:py-2 sm:px-4 sm:py-1 md:font-semibold sm:mt-2 sm:font-medium hover:text-white"
                    loading={buttonLoading}
                  >
                    {t('Sign Up')}
                  </Button>
                )}
              </div>
            </form>
          </div>
          <div className="md:hidden mt-2 text-xs text-center dark:text-dark-text-fill">
            {t('First time here?')}
            <Link to="/signup/org" className="mx-1 text-primary">
              {t('Register')}
            </Link>
            {t('your organization')}
          </div>
        </div>
      </div>
    </div>
  );
}

export default Signup;
