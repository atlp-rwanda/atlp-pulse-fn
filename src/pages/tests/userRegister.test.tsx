/* eslint-disable func-names */
// @ts-nocheck
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { MockedProvider as ApolloProvider } from '@apollo/client/testing';
import UserRegister from '../Organization/UserRegister';

describe('TraineeRatingDashboard Tests', () => {
  const observe = jest.fn();
  const disconnect = jest.fn();
  window.IntersectionObserver = jest.fn(function () {
    this.observe = observe;
    this.disconnect = disconnect;
  });

  it('Renders TraineeRatingDashboard', () => {
    const elem = renderer
      .create(
        <MemoryRouter>
          <ApolloProvider>
            <UserRegister />
          </ApolloProvider>
        </MemoryRouter>,
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
});