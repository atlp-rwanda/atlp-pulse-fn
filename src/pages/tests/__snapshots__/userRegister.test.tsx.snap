// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TraineeRatingDashboard Tests Renders TraineeRatingDashboard 1`] = `
<div
  className="md:flex md:flex-col md:items-center md:justify-center w-full grow text-center py-2 dark:bg-dark-bg bg-gray-100  sm:flex sm:flex-row sm:items-center sm:justify-center"
>
  <div
    className="w-full md:w-3/5 md:p-5 sm:w-full sm:p-2 dark:bg-dark-bg dark:rounded-none "
  >
    <div
      className="py-10 sm:py-8 w-full m-auto  md:shadow-xl rounded-xl"
    >
      <h2
        className="text-2xl font-bold text-primary dark:text-dark-text-fill "
      >
        Sign up using
      </h2>
      <div
        className="border-2 w-10 bg-primary border-primary inline-block mb-2"
      />
      <p
        className="text-gray-400 my-3 dark:text-dark-text-fill "
      >
        use your email account
      </p>
      <div
        className="flex flex-col items-center"
      >
        <form
          action="#none"
          className="flex flex-col gap-2"
          data-testid="loginForm"
          onSubmit={[Function]}
        >
          <div
            className="flex justify-between flex-col md:flex-row gap-2"
          >
            <div
              className="bg-gray-100 border border-primary rounded p-2 flex justify-between  dark:bg-dark-bg w-full mr-2"
            >
              <input
                className="bg-gray-100 outline-none text-sm flex-1 text-gray-400 dark:border-white dark:bg-dark-bg dark:text-white "
                data-testid="fName"
                name="firstName"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="First Name"
                type="name"
              />
            </div>
            <div
              className="bg-gray-100 border border-primary rounded p-2 flex justify-between dark:bg-dark-bg w-full"
            >
              <input
                className="bg-gray-100 outline-none text-sm flex-1 text-gray-400 dark:border-white dark:bg-dark-bg dark:text-white "
                data-testid="lName"
                name="lastName"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Last Name"
                type="name"
              />
            </div>
          </div>
          <div
            className="flex justify-between flex-col md:flex-row gap-2"
          >
            <div
              className="grouped-input flex items-center  w-full  rounded-md mr-2"
            >
              <input
                className="text-gray-400 border border-primary py-2 dark:bg-dark-bg rounded outline-none px-5 font-sans text-xs w-full h-[38px] "
                data-placeholder="Date of birth"
                id="date-placeholder"
                max="2024-03-12"
                name="dateOfBirth"
                onBlur={[Function]}
                onChange={[Function]}
                required={true}
                type="date"
              />
            </div>
            <div
              className="my-react-select-container css-b62m3t-container"
              onKeyDown={[Function]}
            >
              <span
                className="css-1f43avz-a11yText-A11yText"
                id="react-select-2-live-region"
              />
              <span
                aria-atomic="false"
                aria-live="polite"
                aria-relevant="additions text"
                className="css-1f43avz-a11yText-A11yText"
              />
              <div
                className="my-react-select__control css-1an72ab-control"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <div
                  className="my-react-select__value-container css-1fdsijx-ValueContainer"
                >
                  <div
                    className="my-react-select__placeholder css-1jqq78o-placeholder"
                    id="react-select-2-placeholder"
                  >
                    Select gender
                  </div>
                  <div
                    className="my-react-select__input-container css-qbdosj-Input"
                    data-value=""
                  >
                    <input
                      aria-autocomplete="list"
                      aria-describedby="react-select-2-placeholder"
                      aria-expanded={false}
                      aria-haspopup={true}
                      autoCapitalize="none"
                      autoComplete="off"
                      autoCorrect="off"
                      className="my-react-select__input"
                      disabled={false}
                      id="react-select-2-input"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      role="combobox"
                      spellCheck="false"
                      style={
                        Object {
                          "background": 0,
                          "border": 0,
                          "color": "inherit",
                          "font": "inherit",
                          "gridArea": "1 / 2",
                          "label": "input",
                          "margin": 0,
                          "minWidth": "2px",
                          "opacity": 1,
                          "outline": 0,
                          "padding": 0,
                          "width": "100%",
                        }
                      }
                      tabIndex={0}
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  className="my-react-select__indicators css-1hb7zxy-IndicatorsContainer"
                >
                  <span
                    className="my-react-select__indicator-separator css-1u9des2-indicatorSeparator"
                  />
                  <div
                    aria-hidden="true"
                    className="my-react-select__indicator my-react-select__dropdown-indicator css-1xc3v61-indicatorContainer"
                    onMouseDown={[Function]}
                    onTouchEnd={[Function]}
                  >
                    <svg
                      aria-hidden="true"
                      className="css-tj5bde-Svg"
                      focusable="false"
                      height={20}
                      viewBox="0 0 20 20"
                      width={20}
                    >
                      <path
                        d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="bg-gray-100 w-70 p-2 flex justify-between  dark:bg-dark-bg border border-primary rounded xs:w-full gap-2"
          >
            <svg
              className="text-gray-400 "
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M464 64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V112c0-26.51-21.49-48-48-48zm0 48v40.805c-22.422 18.259-58.168 46.651-134.587 106.49-16.841 13.247-50.201 45.072-73.413 44.701-23.208.375-56.579-31.459-73.413-44.701C106.18 199.465 70.425 171.067 48 152.805V112h416zM48 400V214.398c22.914 18.251 55.409 43.862 104.938 82.646 21.857 17.205 60.134 55.186 103.062 54.955 42.717.231 80.509-37.199 103.053-54.947 49.528-38.783 82.032-64.401 104.947-82.653V400H48z"
              />
            </svg>
            <input
              className="bg-gray-100 outline-none text-sm flex-1 text-gray-400 dark:border-white dark:bg-dark-bg dark:text-white  "
              data-testid="email"
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Email"
              readOnly={true}
              type="email"
            />
          </div>
          <div
            className="bg-gray-100  p-2 flex items-center border border-primary rounded   dark:bg-dark-bg xs:w-full md:w-70 gap-2"
          >
            <svg
              className="text-gray-400 "
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 17c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm6-9h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM8.9 6c0-1.71 1.39-3.1 3.1-3.1s3.1 1.39 3.1 3.1v2H8.9V6zM18 20H6V10h12v10z"
              />
            </svg>
            <input
              className="bg-gray-100 outline-none text-sm flex-1 text-gray-400 dark:border-white dark:bg-dark-bg dark:text-white"
              data-testid="password"
              name="password"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />
            <div
              className="text-gray-400 cursor-pointer onClick= {()=> handleShowPassword}"
            >
              <svg
                fill="none"
                height="1em"
                onClick={[Function]}
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M17.94 17.94A10.07 10.07 0 0 1 12 20c-7 0-11-8-11-8a18.45 18.45 0 0 1 5.06-5.94M9.9 4.24A9.12 9.12 0 0 1 12 4c7 0 11 8 11 8a18.5 18.5 0 0 1-2.16 3.19m-6.72-1.07a3 3 0 1 1-4.24-4.24"
                />
                <line
                  x1="1"
                  x2="23"
                  y1="1"
                  y2="23"
                />
              </svg>
            </div>
          </div>
          <div
            className="flex w-64  rounded mb-5 mt-5 items-center "
          >
            <div
              className=" text-sm mr-1"
            >
              Already have an account?
            </div>
            <a
              className="text-xs text-primary "
              href="/login/org"
              onClick={[Function]}
            >
              Log in
            </a>
          </div>
          <div
            className="w-full justify-center"
          >
            <button
              className="btn transparentbtn md border-2 hover:bg-primary inline-block rounded-full lg:px-12 lg:py-2 sm:px-4 sm:py-1 md:font-semibold sm:mt-2 sm:font-medium hover:text-white"
              disabled={false}
              type="submit"
            >
              Sign Up
            </button>
          </div>
        </form>
      </div>
      <div
        className="md:hidden mt-2 text-xs text-center dark:text-dark-text-fill"
      >
        First time here?
        <a
          className="mx-1 text-primary"
          href="/signup/org"
          onClick={[Function]}
        >
          Register
        </a>
        your organization
      </div>
    </div>
  </div>
</div>
`;
